### ISSUE TITLE
Toggle enable_secure_boot property within Node Pools

### ISSUE THREAD

**Issue by abhilashquantiphi** [terraform-google-modules/terraform-google-kubernetes-engine#1683]
### TL;DR

We have a GKE cluster created through beta-private-cluster-update-variant module. I have been trying to use an explicit bool type variable to toggle  `enable_secure_boot` property for different environments.

### Expected behavior

Addition of an explicit variable to tweak/toggle `enable_secure_boot` property with default value set to `false` should not affect the remote state of existing GKE node pool resources.

### Observed behavior

 `enable_secure_boot` flag is a node pool property that has a behaviour to create and destroy node pools when it is toggled. My understanding on [README.md](https://github.com/terraform-google-modules/terraform-google-kubernetes-engine/blob/v26.1.1/modules/beta-private-cluster-update-variant/README.md#:~:text=Optional-,enable_secure_boot,-Secure%20Boot%20helps) is that default value for enable_secure_boot should be `false` until it is specified as `true`(bool) for secure boot enablement.

In Terraform Plan, I discovered an issue that even after setting default value for `enable_secure_boot` to `false` explicitly (mentioned in test environment TF configuration) within calling module; terraform is making the force replacement for the existing node pools which is not an expected behaviour.

### Terraform Configuration

```hcl
locals {
  demo_node_pool = {
    name               = "demo-node-pool-xxxx"
    machine_type       = "n1-standard-4"
    enable_secure_boot = var.enable_secure_boot
  }
}

variable "enable_secure_boot" {
  type        = bool
  description = "Enable/Disable secure boot on node pools"
  default     = false
}

# gke module
module "gke" {
  source     = "terraform-google-modules/kubernetes-engine/google//modules/beta-private-cluster-update-variant"
  project_id = "project-id-####"
  node_pools = local.demo_node_pool
}

# Calling modules
module "dev_env" {
  source             = "../gke"
  enable_secure_boot = true
}

module "test_env" {
  source             = "../gke"
  enable_secure_boot = false
}
```

### Terraform Version

```sh
tf version
Terraform v1.3.7
on linux_amd64

---

# module.test_env.module.gke.google_container_node_pool.pools["gke-np-xx"] must be replaced
+/- resource "google_container_node_pool" "pools" {
      ~ id                          = "projects/demo-node-pool-xxxx" -> (known after apply)
      ...
    }
  +/- resource "random_id" "name" {
      ~ keepers     = { # forces replacement
          ~ "enable_secure_boot"          = "" -> "false"
            # (18 unchanged elements hidden)
        }
        # (2 unchanged attributes hidden)
    }

  # module.test_env.module.gke.random_id.name["gke-np-xx"] must be replaced
+/- resource "random_id" "name" {
      ~ keepers     = { # forces replacement
          ~ "enable_secure_boot"          = "" -> "false"
            # (18 unchanged elements hidden)
        }
        # (2 unchanged attributes hidden)
    }

  # module.test_env.module.gke.random_shuffle.available_zones must be replaced
+/- resource "random_shuffle" "available_zones" {
      ~ id           = "-" -> (known after apply)
      ~ input        = [
          - "us-central1-a",
          - "us-central1-b",
          - "us-central1-c",
        ] -> (known after apply) # forces replacement
      ~ result       = [
          - "us-central1-f",
          - "us-central1-b",
          - "us-central1-c",
        ] -> (known after apply)
        # (1 unchanged attribute hidden)
    }
```
